docs/explanation/priority-inversion-inheritance.rst

```diff
@@ -0,0 +1,67 @@
+The present and the following section will introduce unbounded priority inversion and the need for priority inheritance.
+
+Unbounded priority inversion
+----------------------------
+
+Let's begin with priority inversion by looking at the diagram below.
+Three tasks, L, M, and H, with varying priority levels, low, medium and high, are present in the kernel and about to contest for CPU access.
+
+The low-priority task L runs until it takes a lock; in the diagram below, the blue bar turns purple.
+After acquiring it, task L holds the lock and begins modifying some critical sections within the kernel. 
+Once the higher-priority task H appears, it preempts task L and starts to run.
+At this point, task H would like to acquire the same lock task L is holding.
+As it can't do so, the higher-priority task H goes to sleep and waits for the lower-priority task L to release the lock.
+Task L thus continues running while Task H is sleeping.
+In such a setup, priority inversion can occur if a medium-priority task M comes along and preempts task L.
+Once task M starts running, the high-priority task H will potentially wait for an unbounded amount of time, preventing it from doing work in a critical kernel section.
+Improving the flexibility to preempt tasks executing within the kernel would thus help guarantee an upper time boundary.
+
+.. figure:: priority-inversion-inheritance/unbounded-priority-inversion.svg
+    :width: 100%
+    :align: center
+    :alt: Unbounded priority inversion
+
+    Unbounded priority inversion: A high-priority task being indefinitely delayed by a medium-priority task due to a lack of proper priority management.
+
+
+In this specific example, task M finishes running and releases the CPU -- where the horizontal bar turns from green to red in the drawing -- allowing task L to start running again while still holding the lock.
+Only once task L releases it, task H will wake up and acquire the lock, starting its work within the critical section.
+
+Priority inversion occurred on the Mars Rover, and it is a critical challenge for developers and engineers working with real-time systems. With unbounded priority inversion, the need for priority inheritance becomes clear.
+
+Priority Inheritance
+--------------------
+
+A real-time Linux kernel resolves the unbounded latencies of priority inversion via priority inheritance. 
+
+The diagram below helps illustrate the mechanism.
+As before, the low-priority task L starts running and acquires the lock.
+Similarly to the previous scenario, task H wakes up and starts running, but it is soon blocked while attempting to get the lock.
+
+The high-priority task H wants to take the same lock held by the low-priority task L.
+Differently than in the priority inversion's case, and instead of H going to sleep and waiting, priority inheritance occurs, with L acquiring H's priority.
```

Comment:
```suggestion
In this case, instead of H going to sleep and waiting, priority inheritance occurs, with L acquiring H's priority.
```

This is a suggestion, please feel free to leave as is, but the sentence didn't seem to flow very well IMO.

Thanks.